src/main/java/org/apache/sling/api/resource/SyntheticResource.java

```diff
@@ -112,6 +116,17 @@ public String getResourceSuperType() {
         return resourceResolver;
     }
 
+    /**
+     * Exposes the value map containing the single property {@value ResourceResolver#PROPERTY_RESOURCE_TYPE}.
+     */
+    @Override
+    public <AdapterType> AdapterType adaptTo(Class<AdapterType> type) {
```

Comment:
Isn't this a potentially breaking change? The AdapterManager might create a value map (super.adaptTo())-

Reply:
Indeed, fixed with https://github.com/apache/sling-org-apache-sling-api/pull/59/commits/f07e260a78aee73ad416ee82f8392554bb18fada.